File: easy/squares-of-a-sorted-array.py
"""
- square the value and insert to position on array
"""
def sortedSquares(nums):
    left = 0
    right = len(nums) - 1
    res = []
    for i in range(len(nums)):
        if abs(nums[left]) >= abs(nums[right]):
            res.insert(0, nums[left] ** 2)
            left = left + 1
        else:
            res.insert(0, nums[right] ** 2)
            right = right - 1
    return res

print (sortedSquares([-4,-1,0,3,10]))

File: easy/squares-of-a-sorted-array.cpp
vector<int> sortedSquares(vector<int>& nums) {
    /*
    [-5,-1,0,3,4]
    [16, 25]
    i = 0
    j = 4
    k = 4

    if
    */
    int n = nums.size();
    int i =  0, j = n - 1;
    vector<int> res = vector<int>(n,-1);
    for(int k=n-1;k>=0;--k){
        if(abs(nums[i])>=abs(nums[j])){
            res[k] = nums[i]*nums[i];
            i++;
        }else{
            res[k] = nums[j]*nums[j];
            j--;
        }
    }
    return res;
}